idaes/power_generation/unit_models/soec_design.py

```diff
@@ -0,0 +1,621 @@
+from idaes.core.util.exceptions import ConfigurationError, InitializationError
+import idaes.logger as idaeslog
+
+
+@declare_process_block_class(
+    "SoecDesign", doc="Simple SOEC model for process design.")
+class SoecDesignData(UnitModelBlockData):
+    """Simple 0D SOEC model. This model can be used to develop a flowsheet at the
+    design point or for the basis of a surrogate. For off-design applications
+    a more detailed model is required."""
+    CONFIG = UnitModelBlockData.CONFIG(implicit=True)
+    CONFIG.declare(
+        "oxygen_side_package",
+        ConfigValue(
+            domain=is_physical_parameter_block,
+            description="Property package for the oxygen side.",
+            doc=(
+                "Property package for the oxygen side, using "
+                "idaes.power_generation.properties.natural_gas_PR is "
+                "strongly recomended, either Peng-Robinson or Ideal is okay"
+            )
+        )
+    )
+    CONFIG.declare(
+        "hydrogen_side_package",
+        ConfigValue(
+            domain=is_physical_parameter_block,
+            description="Property package for the hydrogen side.",
+            doc=(
+                "Property package for the hydrogen side, using "
+                "idaes.power_generation.properties.natural_gas_PR is "
+                "strongly recomended, either Peng-Robinson or Ideal is okay"
+            )
+        )
+    )
+    CONFIG.declare(
+        "oxygen_side_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Property package arguments for the oxygen side.",
+            doc="Property package arguments for the oxygen side."
+        )
+    )
+    CONFIG.declare(
+        "hydrogen_side_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Property package arguments for the hydrogen side.",
+            doc="Property package arguments for the hydrogen side."
+        )
+    )
+    CONFIG.declare(
+        "reaction_eos",
+        ConfigValue(
+            default=EosType.PR,
+            description="Reaction properties equation of state.",
+            doc=(
+                "Reaction properties equation of state in: "
+                "{EosType.PR, EosType.IDEAL}."
+            )
+        )
+    )
+    CONFIG.declare(
+        "has_heat_transfer",
+        ConfigValue(
+            default=False,
+            domain=Bool,
+            description="Heat transfer term construction flag",
+            doc="Indicates whether the SOEC is adiabatic. Default=False (adiabatic)"
+        )
+    )
+
+    def build(self):
+        """Construct a unit model data block"""
+        super().build()
+
+        # Do a little validation.
+        # The hydrogen side needs water and hydrogen
+        if set(self.config.hydrogen_side_package.component_list) != {"H2", "H2O"}:
+            raise ConfigurationError(
+                "SOEC hydrogen side must contain exactly H2 and H2O"
```

Comment:
Just an observation, but this is a very specific set of components, so I would almost suggest we call this `H2SOEC`. It is also a prime candidate for the new "devel" folder due to being fairly specific.

Reply:
I'm not sure.  It's a specific unit for some power generation like task, but hopefully it can be finished.  I don't want it to be in perpetual development.  Does devel mean in development and not fully supported, or is it anything that's not core and totally generic?

Reply:
That is still up for debate (i.e. we might need two separate folders).